irksome/base_time_stepper.py

```diff
@@ -137,7 +137,10 @@ def advance(self):
         Note: overwrites the value `u0`."""
 
         push_parent(self.u0.function_space().dm, self.stages.function_space().dm)
-        self.solver.solve(bounds=self.stage_bounds)
+        if self.stage_bounds is None:
+            self.solver.solve()
+        else:
+            self.solver.solve(bounds=self.stage_bounds)
```

Comment:
Only needed until proper fix is in Firedrake
```suggestion
        self.solver.solve(bounds=self.stage_bounds)
```